src/middlewared/middlewared/plugins/boot_/environments.py

```diff
@@ -0,0 +1,189 @@
+import datetime
+import errno
+import json
+import os
+import subprocess
+
+from middlewared.api import api_method
+from middlewared.api.current import (
+    BootEnvironmentActivateArgs,
+    BootEnvironmentActivateResult,
+    BootEnvironmentCloneArgs,
+    BootEnvironmentCloneResult,
+    BootEnvironmentDestroyArgs,
+    BootEnvironmentDestroyResult,
+    BootEnvironmentEntry,
+    BootEnvironmentKeepArgs,
+    BootEnvironmentKeepResult,
+)
+from middlewared.service import filterable_api_method, Service, private
+from middlewared.service_exception import CallError, ValidationError
+from middlewared.utils import filter_list
+from middlewared.utils.size import format_size
+
+
+def run_zectl_cmd(cmd):
+    try:
+        cmd.remove("zectl")
```

Comment:
Looks like a hack. Can we just do `subprocess.run(["zectl"] + cmd)` instead and not change `cmd` at all?

Reply:
fixed